### ISSUE TITLE
Paste JSON or XML as classes: Feature missing or requirements not listed

### ISSUE THREAD

**Issue by skycommand** [MicrosoftDocs/visualstudio-docs#9031]
Hello.

According to the "[Paste JSON or XML as classes](https://learn.microsoft.com/en-us/visualstudio/ide/reference/paste-json-xml?view=vs-2022)" page of Microsoft Learn, the "**Edit**" menu in Visual Studio should have a submenu called "**Paste Special**", which contains two items: "**Paste JSON As Classes**" and "**Paste XML As Classes**".

I have the latest version of Visual Studio installed, but the aforementioned submenu is missing from it.
| Actual | Expected |
|--------|----------|
| ![Actual](https://user-images.githubusercontent.com/17097175/223364251-311a1d20-2bf2-4e68-9097-509f50930744.jpg) | ![Expected](https://learn.microsoft.com/en-us/visualstudio/ide/reference/media/paste-json-xml-class-lrg.png?view=vs-2017) |

Possible causes:

- Microsoft has discontinued this feature
- This feature has prerequisites that the documentation page has failed to mention

Please advise.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 39b68cc1-e6cb-11cc-0830-1ef1439ef46f
* Version Independent ID: 39b68cc1-e6cb-11cc-0830-1ef1439ef46f
* Content: [Paste JSON or XML as classes - Visual Studio (Windows)](https://learn.microsoft.com/en-us/visualstudio/ide/reference/paste-json-xml?view=vs-2022)
* Content Source: [docs/ide/reference/paste-json-xml.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/main/docs/ide/reference/paste-json-xml.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-general**
* GitHub Login: @TerryGLee
* Microsoft Alias: **tglee**

**Comment by TerryGLee** [comment]
@skycommand Thanks for letting us know about your experience! We added the following additional information to the page, based on your feedback: 

"If you don't see the **Paste Special** option on the **Edit** menu, make sure that you select the program file for your app. For example, for a C# app, select the **Program.cs** file in [Solution Explorer](https://learn.microsoft.com/visualstudio/ide/use-solution-explorer)."

We'll close this issue now, with our thanks to you for your help with improving the Visual Studio documentation!

**Comment by skycommand** [comment]
@TerryGLee

I just tried your suggestion. It still doesn't work.

I've been reading comments from all over the Internet. Developers seem to believe that installation of a certain workload is required.

**Comment by TerryGLee** [comment]
@skycommand Thanks for the additional input. I'll investigate further. But to help with that, if you're amenable, would you tell me which workloads you have installed?

**Comment by skycommand** [comment]
Certainly. My workloads are:

- Core Editor
- .NET desktop development
    - .NET desktop development tools
    - .NET Framework 4.7.2 development tools
    - C# and Visual Basic
    - Development tools for .NET
    - .NET Framework 4.8 development tools
    - .NET Profiling tools
    - IntelliCode
    - Just-in-Time debugger
    - Blend for Visual Studio
    - Windows App SDK C# Templates
- Single-project MSIX Packaging Tools for Visual Studio 2022

**Comment by skycommand** [comment]
@TerryGLee Hello again. Follow-up question: How about yours?

**Comment by TerryGLee** [comment]
@skycommand I uninstalled _all_ workloads and components, which removed the **Edit** > **Paste Special** menu item. I then installed the **.NET desktop development** workload and accepted it as is without modifying any of its default components, and the **Edit** > **Paste Special** menu item appears with its **Paste JSON As Classes** and **Paste XML As Classes** options available.

So, for Visual Studio 2022 version 17.5.2, that looks like this:

![image](https://user-images.githubusercontent.com/23244653/225164275-16f47702-beb2-4d5d-bbaa-29d8c8f109bb.png)


Please try the same and let me know what you experience?

**Comment by skycommand** [comment]
@TerryGLee I confirm. Both **Paste JSON As Classes** and **Paste XML As Classes** appear when I install **Live Share** and disappear when I remove Live Share. This component is part of many other workloads.

**Comment by TerryGLee** [comment]
@skycommand Apologies for the delayed response. I checked in with the engineering team and have since followed through with their request that we document the workloads that ensure that the **Paste JSON As Classes** and **Paste XML As Classes** commands appear on the **Edit** menu. Those workloads include the following, where only one need be installed to provide the functionality:

- **ASP.NET and web development**
- **Azure development**
- **.NET desktop development**
 
I've added that information to the Tip note beneath the first screenshot on the page, and so I'll close this issue now, along with extending to you our thanks for your help in improving the Visual Studio documentation!

**Comment by StingyJack** [comment]
@TerryGLee  - I dont see these options anymore either and have all those things installed. They were present and have disappeared. This is probably an update related bug.

**Comment by skycommand** [comment]
@StingyJack Please try this:

1. Run Visual Studio Installer
2. Click on "Modify"
3. From the "individual components" tab, find and install the following:
   1. "JavaScript and TypeScript language support"
   2. "C# and Visual Basic"
   3. "C# and Visual Basic Roslyn compilers"

Alternatively, you can just install "Live Share," which installs all of those.

**Comment by StingyJack** [comment]
Got all those already =D
![image](https://user-images.githubusercontent.com/15381181/232260419-960e80a0-8688-4b8e-9645-ea5ac6b34784.png)
![image](https://user-images.githubusercontent.com/15381181/232260423-3574867d-d0d4-423d-ad5b-e5018837c1ea.png)
![image](https://user-images.githubusercontent.com/15381181/232260427-49a806f6-bf76-46bb-aa0c-18e400825b60.png)

Also mentioned while I was researching this was this one, but I already have it installed...
![image](https://user-images.githubusercontent.com/15381181/232260451-4c8e0365-4327-448b-af42-bbfd2112a9ca.png)

I [Reported a Problem to the developer community for this](https://developercommunity.visualstudio.com/t/Paste-JSON-as-Classes-is-missing-from/10339080), hopefully someone there has an idea of what to do to fix it.

**Comment by danworley** [comment]
The Paste Special option has somehow disappeared from my Visual Studio 2022 Edit menu as well. I was working on a vsix extension when i noticed it had disappeared. Not sure if it's related or not. I've tried uninstalling all those workloads mentioned above then re-installing them, and its still missing. I ended up just installing an extension called PasteFromJsonToClassObject in order to achieve the same results. Really annoying though.
